arabic/unpacked/arabic.js

```diff
@@ -0,0 +1,448 @@
+  MathJax.Hub.Config({
+    'HTML-CSS': {
+      styles: {
+        '.MathJax .mfliph': {
+          'display': 'inline-block !important',
+          '-moz-transform': 'scaleX(-1)',
+          '-webkit-transform': 'scaleX(-1)',
+          '-o-transform': 'scaleX(-1)',
+          'transform': 'scaleX(-1)',
+          '-ms-filter': 'fliph',
+          'filter': 'fliph'
+        },
+        '.MathJax .mar': {
+          'font-style': 'normal !important'
+        },
+        '.MathJax .mar > span': {
+          'font-style': 'normal !important'
+        }
+      }
+    }
+  });
+});
+
+
+MathJax.Hub.Register.StartupHook('HTML-CSS Jax Ready', function () {
+  MathJax.Hub.Register.StartupHook('Arabic TeX Ready', function () {
+    var MML = MathJax.ElementJax.mml;
+
+    var makeElementFlippable = function (name) {
+      var originalToHTML = MML[name].prototype.toHTML;
+
+      MML[name].Augment({
+        toHTML: function () {
+          var element = originalToHTML.apply(this, arguments);
+
+          if (this.arabicFlipH) {
+            var flipElement = document.createElement('span');
+
+            var className = ' mfliph';  // Keep the leading space
+
+            if ('ar' === this.arabicFontLang) {
+              className += ' mar';  // Keep the leading space
+            }
+
+            flipElement.className = className;
+
+            if (element.firstChild) {
+              while (element.firstChild) {
+                flipElement.appendChild(element.firstChild);
+              }
+            }
+
+            element.appendChild(flipElement);
+          }
+
+          return element;
+        }
+      });
+    };
+
+    [
+      'mfrac',
+      'mi',
+      'mn',
+      'mo',
+      'mrow',
+      'ms',
+      'msqrt',
+      'msubsup',
+      'mroot',
+      'mtext'
+    ].forEach(makeElementFlippable);
+
+    MathJax.Hub.Register.StartupHook('HTML-CSS mtable Ready', function () {
+      makeElementFlippable('mtable');
+
+      MathJax.Hub.Startup.signal.Post('Arabic mtable Ready');
+    });
+
+
+    MathJax.Hub.Startup.signal.Post('Arabic Ready');
+  });
+});
+
+MathJax.Hub.Register.StartupHook('Arabic TeX Startup', function () {
+  var Text = MathJax.Extension.Arabic.Text;
+
+  MathJax.Hub.CombineConfig("Arabic", {
+      identifiersMap: {
+        // Math functions
+        'sin': 'جا',
+        'cos': 'جتا',
+        'tan': 'ظا',
+        'cot': 'ظتا',
+        'sec': 'قا',
+        'csc': 'قتا',
+        'log': 'لو'
+      },
+      // Limits
+      operatorsMap: {
+        'lim': 'نهــا'
+      }
+    });
+
+  MathJax.Extension.Arabic.config = MathJax.Hub.CombineConfig('Arabic', {
+    dict: {
+      // A macros to force English zero in both languages
+      "Zero": ["zero", Text('0', 'صفر')],  // Better localized Zero
+      "Radius": ["radius", Text('r', 'نق')],  // Circle radius
+      "Area": ["Area", Text('A', 'م')]  // Area of circles and other stuff
+    }
+  });
+});
```

Comment:
I'm a little confused by the multiple calls to `CombineConfig()`.  It is not really meant to be used more than once, but you can do it, as you have seen, but setting `MathJax.Extension.Arabic.config each time.

But I really don't see why you need to break up the configuration into so many pieces.  For example, why not just do

    MathJax.Extension.Arabic.config = MathJax.Hub.CombineConfig('Arabic', {
      identifiersMap: {
        ...
      },
      operatorsMap: {
        ...
      },
      dict: {
      ...
      }
    });

with all the items together in one call?   Why bother with the overhead of multiple calls?

Similarly, I don't see the need for lines 28-55 and 174-194 to be separate as well.  I assume it is because you are using separate files in your development process, but I would hope that that development process would not produce an awkward final version.  You could probably do without the internal Arabic TeX Startup signal if you structured this differently.